app/Views/Home/Req.cshtml

```diff
@@ -15,17 +15,17 @@
     <div class="field__body">
       <input class="field__input" value="@HttpContextAccessor.HttpContext.Session.GetString("access_token")" required readonly />
       <a class="button" href="@HttpContextAccessor.HttpContext.Session.GetString("BaseUrl")/refresh_token/">Refresh Token</a>
+      <input type="hidden" id="hidden_expires_at" value="@HttpContextAccessor.HttpContext.Session.GetString("expires_at")">
     </div>
-  </div> 
+  </div>
 
   <script>
     // Token expiry timer
     (function() {
+      var expires = document.getElementById('hidden_expires_at').value + "000"; // add milliseconds

```

Comment:
If you use the `expires_at` only to calculate the timer, why don't you just use it here without adding an extra hidden field? 
```suggestion
      var expires = "@HttpContextAccessor.HttpContext.Session.GetString("expires_at")" + "000";
```

Reply:
Thanks Tim, it really works ;) I changed it.

Reply:
Of course it does. 😉